astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -1048,60 +1059,64 @@ def test_int_out_of_range(parallel):
     expected = Table([[12.3, 10.*imax], [10.*imax, 4.56e8]],
                      names=('A', 'B'))
 
-    table = ascii.read(text, format='basic', guess=False,
+    table = ascii.read(text, format='basic', guess=guess,
                        fast_reader={'parallel': parallel})
     assert_table_equal(table, expected)
-    table = ascii.read(text, format='basic', guess=False, fast_reader=False)
+    table = ascii.read(text, format='basic', guess=guess, fast_reader=False)
     assert_table_equal(table, expected)
 
 
+@pytest.mark.parametrize("guess", [True, False])
 @pytest.mark.parametrize("parallel", [
     pytest.param(True, marks=pytest.mark.xfail(os.name == 'nt', reason="Multiprocessing is currently unsupported on Windows")),
     False])
-def test_fortran_reader(parallel):
+
+def test_fortran_reader(parallel, guess):
```

Comment:
@dhomeier - can you help me out and explain what is happening with these test diffs?  At first glance they seem unrelated to the goal of this PR which is improved exception consistency and using all options in the fast reader for guessing.

Are the new tests strictly a superset and/or more stringent than the original?

Reply:
The `guess` parameter was introduced to make them a superset as far as possible, i.e. wherever the original was calling `read()` with `guess=False` (because `True` would not work), it is now testing both options. So it is related to item 3 of the PR (which did sneak in as things were evolving, admitted).